### PR TITLE
Fix loading vhost_vsock module when VMs are not supported

### LINKED ISSUE #11794: `vhost_vsock` is loaded even when VMs are not supported
# Issue description

On machines where VMs are not supported (i.e. administratively disabled by blocking `kvm`), the module `vhost_vsock` is still loaded, presumably by [lxd/sys/os.go](https://github.com/lxc/lxd/blob/master/lxd/sys/os.go#L185).

This `vhost_vsock` module and its dependencies are AFAIK only useful for VMs so it would be ideal to no load them if VMs are not supported by the LXD host.

# Steps to reproduce

```
# Disable KVM support
cat << EOF > /etc/modprobe.d/blacklist-kvm.conf
install kvm /bin/true
install kvm_amd /bin/true
install kvm_intel /bin/true
EOF
update-initramfs -uk all
reboot
```

```
# Confirm LXD cannot use VMs:
$ for w in $(lxc warning list -f csv -c u); do lxc warning show "$w"; done | grep -wF KVM
last_message: KVM support is missing (no /dev/kvm)

# List vhost modules
$ lsmod | grep -E '^Module|vhost'
Module                  Size  Used by
vhost_vsock            28672  0
vmw_vsock_virtio_transport_common    45056  1 vhost_vsock
vhost                  53248  1 vhost_vsock
vhost_iotlb            16384  1 vhost
vsock                  49152  2 vmw_vsock_virtio_transport_common,vhost_vsock
```

`vhost_vsock` is not used (`Used by 0`) but caused the other `vhost` related modules to be loaded.

# Information to attach

```
$ snap list lxd; uname -a
Name  Version        Rev    Tracking      Publisher   Notes
lxd   5.0.2-838e1b2  24322  5.0/stable/…  canonical✓  -
Linux xeon 5.19.0-43-generic #44~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Mon May 22 13:39:36 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
```

### PR DESCRIPTION
**Pull request by MusicDin**
Fixes #11794

### CODE REVIEW

File: lxd/instance/drivers/driver_qemu.go [hunk 1]
```diff
@@ -7873,6 +7881,14 @@ func (d *qemu) checkFeatures(hostArch int, qemuPath string) (map[string]any, err
 		features["vhost_net"] = struct{}{}
 	}
 
+	vsockID, err := vsock.ContextID()
+	if err != nil || vsockID > 2147483647 {
+		// Fallback to the default ID for a host system
+		features["vhost_vsock"] = 2
```
Comment (tomponline) [inline lxd/instance/drivers/driver_qemu.go:7887]: Can use `vsock.Host` constant here to avoid using a "magic number"?

**Comment by MusicDin** [comment]
@tomponline To me the tests look OK, but please take another look. Otherwise if there is no other comments it can be merged.

**Review by tomponline — APPROVED** [review]
LGTM! Thanks

**Comment by MusicDin** [comment]
> @MusicDin the test suite does not check VM functionality. Can you confirm VMs working OK for you and you can login via vsock using `lxc shell <instance>`?

@tomponline For me the following works just fine:
```sh
lxc launch ubuntu:22.04 vm --vm

lxc shell vm
# root@vm:~#

```

**Pull request merged**